app/controllers/application_controller.rb

```diff
@@ -7,12 +7,16 @@ def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
+  def require_user
+    render file: 'public/404', status: :not_found unless current_user
+  end
+
   def generate_flash(resource)
-    error_messages = String.new
+    error_messages = ''
```

Comment:
These seem to be doing the same thing, why did you go with the literal string instead of initializing an empty string?

Reply:
This was a rubucop violation - it recommended a literal string instead of String.new :)